homeassistant/components/goodwe/sensor.py

```diff
@@ -0,0 +1,252 @@
+        state_class=STATE_CLASS_MEASUREMENT,
+        native_unit_of_measurement=TEMP_CELSIUS,
+    ),
+    "Hz": SensorEntityDescription(
+        key="Hz",
+        device_class=DEVICE_CLASS_VOLTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        native_unit_of_measurement=FREQUENCY_HERTZ,
+    ),
+    "%": SensorEntityDescription(
+        key="%",
+        state_class=STATE_CLASS_MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the GoodWe inverter from a config entry."""
+    entities = []
+    inverter = hass.data[DOMAIN][config_entry.entry_id][KEY_INVERTER]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+
+    device_info = DeviceInfo(
+        configuration_url="https://www.semsportal.com",
+        identifiers={(DOMAIN, config_entry.unique_id)},
+        name=config_entry.title,
+        manufacturer="GoodWe",
+        model=inverter.model_name,
+        sw_version=f"{inverter.software_version} ({inverter.arm_version})",
+    )
+
+    # Entity representing inverter itself
+    inverter_entity = InverterEntity(coordinator, device_info, inverter)
+    entities.append(inverter_entity)
+
+    # Individual inverter sensors entities
+    entities.extend(
+        InverterSensor(coordinator, device_info, inverter, sensor)
+        for sensor in inverter.sensors()
+        if not sensor.id_.startswith("xx")
+    )
+
+    async_add_entities(entities)
+
+    # Add services
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_SET_WORK_MODE,
+        {vol.Required(ATTR_WORK_MODE): vol.Coerce(int)},
+        "async_set_work_mode",
+    )
+    platform.async_register_entity_service(
+        SERVICE_SET_ONGRID_BATTERY_DOD,
+        {vol.Required(ATTR_ONGRID_BATTERY_DOD): vol.Coerce(int)},
+        "async_set_ongrid_battery_dod",
+    )
+    platform.async_register_entity_service(
+        SERVICE_SET_GRID_EXPORT_LIMIT,
+        {vol.Required(ATTR_GRID_EXPORT_LIMIT): vol.Coerce(int)},
+        "async_set_grid_export_limit",
+    )
+
+    return True
+
+
+class InverterEntity(CoordinatorEntity, SensorEntity):
+    """Entity representing the inverter instance itself."""
+
+    _MAIN_ENTITY_SENSOR = "ppv"
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        device_info: DeviceInfo,
+        inverter: Inverter,
+    ) -> None:
+        """Initialize the main inverter entity."""
+        super().__init__(coordinator)
+        self.entity_id = f".{DOMAIN}_inverter"
+        self._attr_unique_id = f"{DOMAIN}-{inverter.serial_number}"
+        self._attr_device_info = device_info
+        self._attr_icon = "mdi:solar-power"
+        self._attr_name = "PV Inverter"
+        self._attr_native_unit_of_measurement = POWER_WATT
+        self._attr_extra_state_attributes = {"serial_number": inverter.serial_number}
```

Comment:
Please remove this. We don't allow static info in state attributes unless it's essential for automations.
```suggestion
```

Reply:
Done.